<?xml version="1.0" encoding="utf-8"?>
<launch>
  <arg name="launch_prefix"       default=""/>
  <arg name="rec_tol"             default="0.70"/>
  <arg name="launch_freenect"     default="false"/>
  <arg name="launch_realsense"    default="true"/>
  <arg name="launch_head_cam"     default="false"/>
  <arg name="launch_reconfigure"  default="false"/>
  <arg name="node_name"           default="PersonDetector"/>
  
  <include if="$(arg launch_freenect)"   file="$(find freenect_launch)/launch/freenect.launch">
    <arg name="depth_registration"       value="true"/>
  </include>
  
  <include if="$(arg launch_realsense)"  file="$(find realsense2_camera)/launch/realsense.launch">
    <arg name="camera_name"              value="camera3D"/>
    <!--<arg name="depth_width"              value="640"/>
    <arg name="depth_height"             value="480"/>
    <arg name="rgb_width"                value="640"/>
    <arg name="rgb_height"               value="480"/>--> 
  </include>
  
  <include if="$(arg launch_head_cam)"   file="$(find video_stream_opencv)/launch/my_camera.launch">
    <arg name="video_stream_provider"    value="3"/>
    <arg name="camera_name"              value="head_camera"/>
  </include>
  
  <!-- Load parameters -->
  <rosparam command="load" ns="$(arg node_name)" file="$(find image_utils)/cfg/general_params.yaml"/>
  <rosparam command="load" ns="$(arg node_name)" file="$(find person_detector)/cfg/person_detector_params.yaml"/>
  <param name="$(arg node_name)/device" value="Kinect" if="$(arg launch_freenect)"/>
  <param name="$(arg node_name)/device" value="Realsense" if="$(arg launch_realsense)"/>
  
  <!-- Launch RQT reconfigure -->
  <group if="$(arg launch_reconfigure)">
      <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="rqt_reconfigure" output="screen"/>
  </group>  

  <!-- Person Detector -->
  <node pkg="person_detector" type="person_detector" name="$(arg node_name)" output="screen">
    <param name="command" value="$(arg commands)"/>
  </node>
  
  <!-- Servers -->
  <node pkg="person_detector" type="openFaceServer.py" name="face_recognition_server" output="screen">
     <param name="rec_tol" value="$(arg rec_tol)"/>
  </node>

</launch>
